src/lib.rs

```diff
@@ -0,0 +1,123 @@
+use actix_web::{
+    error,
+    http::{header::ContentType, StatusCode},
+    web, HttpResponse, Responder,
+};
+use derive_more::{Display, Error};
+use sha3::{Digest, Keccak256};
+
+pub mod structs;
+use crate::structs::{AbiMethod, Request, Response, Transaction};
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, Display, Error)]
+pub enum Error {
+    #[display(fmt = "internal error")]
+    InternalError,
+
+    #[display(fmt = "bad request")]
+    BadClientData,
+
+    #[display(fmt = "not found")]
+    NotFound,
+}
+
+impl From<reqwest::Error> for Error {
+    fn from(_: reqwest::Error) -> Self {
+        Error::NotFound
+    }
+}
+
+impl From<serde_json::Error> for Error {
+    fn from(_: serde_json::Error) -> Self {
+        Error::BadClientData
+    }
+}
+
+impl From<actix_web::Error> for Error {
+    fn from(_: actix_web::Error) -> Self {
+        Error::InternalError
+    }
+}
+
+impl error::ResponseError for Error {
+    fn error_response(&self) -> HttpResponse {
+        HttpResponse::build(self.status_code())
+            .insert_header(ContentType::html())
+            .body(self.to_string())
+    }
+
+    fn status_code(&self) -> StatusCode {
+        match *self {
+            Error::InternalError => StatusCode::INTERNAL_SERVER_ERROR,
+            Error::BadClientData => StatusCode::BAD_REQUEST,
+            Error::NotFound => StatusCode::NOT_FOUND,
+        }
+    }
+}
+
+async fn get_txn_input(txn_hash: &String) -> Result<String> {
+    let res = reqwest::get(format!(
+        "https://blockscout.com/eth/mainnet/api?module=transaction&action=gettxinfo&txhash={}",
+        txn_hash
+    ))
+    .await?;
+
+    let res = res.text().await?;
+    let res: Transaction = serde_json::from_str(&res)?;
```

Comment:
```suggestion
    let res: Transaction = res.json().await?;
```
You can convert it directly from response if you add feature `json` into `reqwest` dependency